### ISSUE TITLE
Operator and operand code need refactoring

### ISSUE THREAD

**Issue by bcrochet** [opdev/opcap#190]
Especially in the operator/operand install code, there is a lot of overlap in what these do. Also, the install functions are doing more than they should have responsibility for. 

So, this bug should address:
1. Dedupe the report code and have it use a template (or a marshaler)
2. Dedupe the install code.
3. Make the install functions just have the responsibility of install, and not of creating reports.

**Comment by acmenezes** [comment]
@bcrochet , I guess this issue could be split into two other issues with better explanation and directions for other people to contribute as well. What do you think?

**Comment by bcrochet** [comment]
> @bcrochet , I guess this issue could be split into two other issues with better explanation and directions for other people to contribute as well. What do you think?

Yes. I will look at what we have gotten to so far and spell out where it needs to go.
